src/export/InstanceExporter.ts

```diff
@@ -0,0 +1,88 @@
+import { FSHTank } from '../import/FSHTank';
+import { StructureDefinition } from '../fhirtypes';
+import { Instance } from '../fshtypes';
+import { FHIRDefinitions } from '../fhirdefs';
+import { StructureDefinitionExporter } from '.';
+import { logger } from '../utils/FSHLogger';
+import { setFixedValueOfPath } from '../fhirtypes/common';
+import { InstanceOfNotDefinedError } from '../errors/InstanceOfNotDefinedError';
+
+export type InstanceDefinition = {
+  resourceType: string;
+  instanceName: string;
+  id?: string;
+};
+
+export class InstanceExporter {
+  constructor(public readonly FHIRDefs: FHIRDefinitions, public readonly tank: FSHTank) {}
+
+  private setFixedValues(
+    fshInstanceDef: Instance,
+    instanceDef: InstanceDefinition,
+    instanceOfStructureDefinition: StructureDefinition,
+    structDefExporter: StructureDefinitionExporter
+  ): InstanceDefinition {
+    // All rules will be FixValueRule
+    fshInstanceDef.rules.forEach(rule => {
+      const { fixedValue, pathParts } = instanceOfStructureDefinition.validateValueAtPath(
+        rule.path,
+        rule.fixedValue,
+        structDefExporter.resolve.bind(structDefExporter)
+      );
+
+      setFixedValueOfPath(instanceDef, pathParts, fixedValue);
+    });
+
+    return instanceDef;
+  }
+
+  exportInstance(fshDefinition: Instance): InstanceDefinition {
+    const structDefExporter = new StructureDefinitionExporter(this.FHIRDefs, this.tank);
+    const instanceOfStructureDefinition = structDefExporter.resolve(fshDefinition.instanceOf);
+
+    if (!instanceOfStructureDefinition) {
+      throw new InstanceOfNotDefinedError(
+        fshDefinition.name,
+        fshDefinition.instanceOf,
+        fshDefinition.sourceInfo
+      );
+    }
+
+    let instanceDef: InstanceDefinition = {
+      resourceType: instanceOfStructureDefinition.type, // ResourceType is determined by the StructureDefinition of the type
+      instanceName: fshDefinition.id // This is name of the instance in the FSH
+    };
+
+    // All other values of the instance will be fixedValues explicitly on the FSH Instance
+    instanceDef = this.setFixedValues(
+      fshDefinition,
+      instanceDef,
+      instanceOfStructureDefinition,
+      structDefExporter
+    );
+
+    // Add any set values that were set on the StructDef but not explicitly on the FSH Instance
```

Comment:
In general, those to-do tests look helpful we should definitely be using those. But I do have a task this sprint for implementing this feature, so I'd be ok with it if Julia wants to skip that for now. Of course, if she wants to write some tests for me I wouldn't stop her 😉 .

Reply:
Ah, those todo tests look really cool! I've never used them before. I am thinking it might make sense to hold off on them in this particular case since Nick is working on the task to add this functionality in this sprint, but if either of you feel strongly, I can add them now.

Reply:
If it's going to be addressed in this sprint, then I feel like it's not super necessary to add the todo tests.